sdk/ai/Azure.AI.Agents.Persistent/src/Telemetry/OpenTelemetryScope.cs

```diff
@@ -0,0 +1,1159 @@
+                   { GenAiSystemKey, GenAiSystemValue},
+                   { GenAiEventContent, evnt != null ? JsonSerializer.Serialize(evnt) : null  }
+            };
+
+            scope._activity?.AddEvent(
+                new ActivityEvent(EventNameSystemMessage, tags: messageTags)
+            );
+            return scope;
+        }
+
+        /// <summary>
+        /// Create the instance of OpenTelemetryScope for thread creation.
+        /// This constructor logs request and starts the execution timer.
+        /// </summary>
+        /// <param name="content">The request options used in the call.</param>
+        /// <param name="endpoint">The endpoint being called.</param>
+        public static OpenTelemetryScope StartCreateThread(RequestContent content, Uri endpoint)
+        {
+            if (!s_enableTelemetry)
+            {
+                return null;
+            }
+            // Deserialize the RequestContent into a CreateThreadRequest object
+            var createThreadRequest = DeserializeCreateThreadRequest(content);
+            var scope = new OpenTelemetryScope(OperationNameValueCreateThread, endpoint);
+
+            return scope;
+        }
+
+        /// <summary>
+        /// Create the instance of OpenTelemetryScope for message creation.
+        /// This constructor logs request and starts the execution timer.
+        /// </summary>
+        /// <param name="threadId">The id of the thread to which the message is created.</param>
+        /// <param name="content">The request options used in the call.</param>
+        /// <param name="endpoint">The endpoint being called.</param>
+        public static OpenTelemetryScope StartCreateMessage(string threadId, RequestContent content, Uri endpoint)
+        {
+            if (!s_enableTelemetry)
+            {
+                return null;
+            }
+
+            var createMessageRequest = DeserializeCreateMessageRequest(content);
+            object evnt = null;
+            if (s_traceContent)
+            {
+                try
+                {
+                    // Attempt to deserialize as a string
+                    string message = createMessageRequest.Content.ToString().Trim('"');
+                    evnt = new { content = message, role = createMessageRequest.Role.ToString() };
+                }
+                catch
+                {
+                }
+            }
+            else
+            {
+                evnt = new { role = createMessageRequest.Role.ToString() };
+            }
+
+            var scope = new OpenTelemetryScope(OperationNameValueCreateMessage, endpoint);
+            scope.SetTagMaybe(GenAiThreadIdKey, threadId);
+            ActivityTagsCollection messageTags = new() {
+                   { GenAiSystemKey, GenAiSystemValue},
+                   { GenAiEventContent, evnt != null ? JsonSerializer.Serialize(evnt) : null  }
+            };
+            scope._activity?.AddEvent(
```

Comment:
please check if activity is null and [is sampled in](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.activity.isalldatarequested?view=net-9.0MemoryContent) before creating event content